# 1.List all the cities and the respective number of pending orders in 'India'?

In [1]:
import pandas as pd


In [2]:
df = pd.read_csv('DataCoSupplyChainDataset.csv',engine='python')

In [3]:
    df
    

Type  Days for shipping (real)  Days for shipment (scheduled)  \
0          DEBIT                         3                              4   
1       TRANSFER                         5                              4   
2           CASH                         4                              4   
3          DEBIT                         3                              4   
4        PAYMENT                         2                              4   
...          ...                       ...                            ...   
180514      CASH                         4                              4   
180515     DEBIT                         3                              2   
180516  TRANSFER                         5                              4   
180517   PAYMENT                         3                              4   
180518   PAYMENT                         4                              4   

        Benefit per order  Sales per customer   Delivery Status  \
0               91.250000          314.640015  Advance shipping   
1             -249.089996          311.359985     Late delivery   
2             -247.779999          309.720001  Shipping on time   
3               22.860001          304.809998  Advance shipping   
4              134.210007          298.250000  Advance shipping   
...                   ...                 ...               ...   
180514          40.000000          399.980011  Shipping on time   
180515        -613.770019          395.980011     Late delivery   
180516         141.110001          391.980011     Late delivery   
180517         186.229996          387.980011  Advance shipping   
180518         168.949997          383.980011  Shipping on time   

        Late_delivery_risk  Category Id   Category Name Customer City  ...  \
0                        0           73  Sporting Goods        Caguas  ...   
1                        1           73  Sporting Goods        Caguas  ...   
2                        0           73  Sporting Goods      San Jose  ...   
3                        0           73  Sporting Goods   Los Angeles  ...   
4                        0           73  Sporting Goods        Caguas  ...   
...                    ...          ...             ...           ...  ...   
180514                   0           45         Fishing      Brooklyn  ...   
180515                   1           45         Fishing   Bakersfield  ...   
180516                   1           45         Fishing       Bristol  ...   
180517                   0           45         Fishing        Caguas  ...   
180518                   0           45         Fishing        Caguas  ...   

       Order Zipcode Product Card Id Product Category Id  Product Description  \
0                NaN            1360                  73                  NaN   
1                NaN            1360                  73                  NaN   
2                NaN            1360                  73                  NaN   
3                NaN            1360                  73                  NaN   
4                NaN            1360                  73                  NaN   
...              ...             ...                 ...                  ...   
180514           NaN            1004                  45                  NaN   
180515           NaN            1004                  45                  NaN   
180516           NaN            1004                  45                  NaN   
180517           NaN            1004                  45                  NaN   
180518           NaN            1004                  45                  NaN   

                                            Product Image  \
0            http://images.acmesports.sports/Smart+watch    
1            http://images.acmesports.sports/Smart+watch    
2            http://images.acmesports.sports/Smart+watch    
3            http://images.acmesports.sports/Smart+watch    
4            http://images.acmesports.sports/Smart+watch    
...   

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 180519 entries, 0 to 180518
Data columns (total 53 columns):
 #   Column                         Non-Null Count   Dtype  
---  ------                         --------------   -----  
 0   Type                           180519 non-null  object 
 1   Days for shipping (real)       180519 non-null  int64  
 2   Days for shipment (scheduled)  180519 non-null  int64  
 3   Benefit per order              180519 non-null  float64
 4   Sales per customer             180519 non-null  float64
 5   Delivery Status                180519 non-null  object 
 6   Late_delivery_risk             180519 non-null  int64  
 7   Category Id                    180519 non-null  int64  
 8   Category Name                  180519 non-null  object 
 9   Customer City                  180519 non-null  object 
 10  Customer Country               180519 non-null  object 
 11  Customer Email                 180519 non-null  object 
 12  Customer Fname                

In [5]:
df.describe()


Days for shipping (real)  Days for shipment (scheduled)  \
count             180519.000000                  180519.000000   
mean                   3.497654                       2.931847   
std                    1.623722                       1.374449   
min                    0.000000                       0.000000   
25%                    2.000000                       2.000000   
50%                    3.000000                       4.000000   
75%                    5.000000                       4.000000   
max                    6.000000                       4.000000   

       Benefit per order  Sales per customer  Late_delivery_risk  \
count      180519.000000       180519.000000       180519.000000   
mean           21.974989          183.107609            0.548291   
std           104.433526          120.043670            0.497664   
min         -4274.979980            7.490000            0.000000   
25%             7.000000          104.379997            0.000000   
50%            31.520000          163.990005            1.000000   
75%            64.800003          247.399994            1.000000   
max           911.799988         1939.989990            1.000000   

         Category Id    Customer Id  Customer Zipcode  Department Id  \
count  180519.000000  180519.000000     180516.000000  180519.000000   
mean       31.851451    6691.379495      35921.126914       5.443460   
std        15.640064    4162.918106      37542.461122       1.629246   
min         2.000000       1.000000        603.000000       2.000000   
25%        18.000000    3258.500000        725.000000       4.000000   
50%        29.000000    6457.000000      19380.000000       5.000000   
75%        45.000000    9779.000000      78207.000000       7.000000   
max        76.000000   20757.000000      99205.000000      12.000000   

            Latitude  ...  Order Item Quantity          Sales  \
count  180519.000000  ...        180519.000000  180519.000000   
mean       29.719955  ...             2.127638     203.772096   
std         9.813646  ...             1.453451     132.273077   
min       -33.937553  ...             1.000000       9.990000   
25%        18.265432  ...             1.000000     119.980003   
50%        33.144863  ...             1.000000     199.919998   
75%        39.279617  ...             3.000000     299.950012   
max        48.781933  ...             5.000000    1999.989990   

       Order Item Total  Order Profit Per Order  Order Zipcode  \
count     180519.000000           180519.000000   24840.000000   
mean         183.107609               21.974989   55426.132327   
std          120.043670              104.433526   31919.279101   
min            7.490000            -4274.979980    1040.000000   
25%          104.379997                7.000000   23464.000000   
50%          163.990005               31.520000   59405.000000   
75%          247.399994               64.800003   90008.000000   
max         1939.989990              911.799988   99301.000000   

       Product Card Id  Product Category Id  Product Description  \
count    180519.000000        180519.000000                  0.0   
mean        692.509764            31.851451                  NaN   
std         336.446807            15.640064                  NaN   
min          19.000000             2.000000                  NaN   
25%         403.000000            18.000000                  NaN   
50%         627.000000            29.000000                  NaN   
75%        1004.000000            45.000000                  NaN   
max        1363.000000            76.000000                  NaN   

       Product Price  Product Status  
count  180519.000000        180519.0  
mean      141.232550             0.0  
std       139.732492             0.0  
min         9.990000             0.0  
25%        50.000000             0.0  
50%        59.990002             0.0  
75%       199.990005             0.0  
max      1999.989990             0.0  

[8 r

In [6]:
updated_df =df.drop(columns =['Product Description'])

In [7]:
updated_df.info()
#removed the columns with null values

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 180519 entries, 0 to 180518
Data columns (total 52 columns):
 #   Column                         Non-Null Count   Dtype  
---  ------                         --------------   -----  
 0   Type                           180519 non-null  object 
 1   Days for shipping (real)       180519 non-null  int64  
 2   Days for shipment (scheduled)  180519 non-null  int64  
 3   Benefit per order              180519 non-null  float64
 4   Sales per customer             180519 non-null  float64
 5   Delivery Status                180519 non-null  object 
 6   Late_delivery_risk             180519 non-null  int64  
 7   Category Id                    180519 non-null  int64  
 8   Category Name                  180519 non-null  object 
 9   Customer City                  180519 non-null  object 
 10  Customer Country               180519 non-null  object 
 11  Customer Email                 180519 non-null  object 
 12  Customer Fname                

In [8]:
updated_df.shape

(180519, 52)

In [9]:
duplicate_rows = updated_df[updated_df.duplicated()]
print('number of duplicate rows',duplicate_rows.shape)
#trying to find any duplicate rows if present

number of duplicate rows (0, 52)


No Duplicate rows were found

In [10]:
print(updated_df["Order Status"].unique())

['COMPLETE' 'PENDING' 'CLOSED' 'PENDING_PAYMENT' 'CANCELED' 'PROCESSING'
 'SUSPECTED_FRAUD' 'ON_HOLD' 'PAYMENT_REVIEW']


In [11]:
pending_dataframe = df[df['Order Status'] == 'PENDING']

In [12]:
pending_dataframe
#the dataframe where all the orders are pending

Type  Days for shipping (real)  Days for shipment (scheduled)  \
1       TRANSFER                         5                              4   
11      TRANSFER                         5                              2   
12      TRANSFER                         4                              2   
21      TRANSFER                         5                              4   
22      TRANSFER                         4                              2   
...          ...                       ...                            ...   
180414  TRANSFER                         0                              0   
180458  TRANSFER                         4                              4   
180485  TRANSFER                         4                              4   
180491  TRANSFER                         3                              4   
180516  TRANSFER                         5                              4   

        Benefit per order  Sales per customer   Delivery Status  \
1             -249.089996          311.359985     Late delivery   
11              45.689999          268.760010     Late delivery   
12              21.760000          262.200012     Late delivery   
21              82.300003          304.809998     Late delivery   
22              22.370001          298.250000     Late delivery   
...                   ...                 ...               ...   
180414         132.130005          363.980011  Shipping on time   
180458         105.589996          319.980011  Shipping on time   
180485          11.160000          371.980011  Shipping on time   
180491          33.599998          335.980011  Advance shipping   
180516         141.110001          391.980011     Late delivery   

        Late_delivery_risk  Category Id   Category Name   Customer City  ...  \
1                        1           73  Sporting Goods          Caguas  ...   
11                       1           73  Sporting Goods        Freeport  ...   
12                       1           73  Sporting Goods         Salinas  ...   
21                       1           73  Sporting Goods          Caguas  ...   
22                       1           73  Sporting Goods  Rancho Cordova  ...   
...                    ...          ...             ...             ...  ...   
180414                   0           45         Fishing      Washington  ...   
180458                   0           45         Fishing          Caguas  ...   
180485                   0           45         Fishing       San Pablo  ...   
180491                   0           45         Fishing          Caguas  ...   
180516                   1           45         Fishing         Bristol  ...   

       Order Zipcode Product Card Id Product Category Id  Product Description  \
1                NaN            1360                  73                  NaN   
11               NaN            1360                  73                  NaN   
12               NaN            1360                  73                  NaN   
21               NaN            1360                  73                  NaN   
22               NaN            1360                  73                  NaN   
...              ...             ...                 ...                  ...   
180414           NaN            1004                  45                  NaN   
180458           NaN            1004                  45                  NaN   
180485           NaN            1004                  45                  NaN   
180491           NaN            1004                  45                  NaN   
180516           NaN            1004                  45                  NaN   

                                            Product Image  \
1            http://images.acmesports.sports/Smart+watch    
11           http://images.acmesports.sports/Smart+watch    
12           http://images.acmesports.sports/Smart+watch    
21           http://images.acmesports.sports/Smart+watch    
22           http://images.acmesports.sport

In [28]:
pending_indian_dataframe = pending_dataframe[pending_dataframe['Order Country'] == 'India']
type(pending_dataframe['Order Country'] == 'India')

pandas.core.series.Series

In [14]:
pending_indian_dataframe
# this dataframe conrtains all the pending orders in India

Type  Days for shipping (real)  Days for shipment (scheduled)  \
1       TRANSFER                         5                              4   
21      TRANSFER                         5                              4   
45      TRANSFER                         6                              4   
46      TRANSFER                         5                              2   
382     TRANSFER                         3                              4   
...          ...                       ...                            ...   
180142  TRANSFER                         4                              4   
180164  TRANSFER                         2                              1   
180248  TRANSFER                         5                              2   
180331  TRANSFER                         3                              4   
180405  TRANSFER                         2                              4   

        Benefit per order  Sales per customer   Delivery Status  \
1             -249.089996          311.359985     Late delivery   
21              82.300003          304.809998     Late delivery   
45              86.169998          275.309998     Late delivery   
46             130.580002          272.029999     Late delivery   
382            144.210007          327.750000  Advance shipping   
...                   ...                 ...               ...   
180142          68.040001          377.980011  Shipping on time   
180164         123.190002          351.980011     Late delivery   
180248         115.190002          383.980011     Late delivery   
180331        -103.970001          327.980011  Advance shipping   
180405          81.000000          299.989990  Advance shipping   

        Late_delivery_risk  Category Id   Category Name   Customer City  ...  \
1                        1           73  Sporting Goods          Caguas  ...   
21                       1           73  Sporting Goods          Caguas  ...   
45                       1           73  Sporting Goods          Caguas  ...   
46                       1           73  Sporting Goods       Rochester  ...   
382                      0           73  Sporting Goods          Caguas  ...   
...                    ...          ...             ...             ...  ...   
180142                   0           45         Fishing          Caguas  ...   
180164                   1           45         Fishing          Caguas  ...   
180248                   1           45         Fishing       Las Vegas  ...   
180331                   0           45         Fishing  East Brunswick  ...   
180405                   0           45         Fishing         Phoenix  ...   

       Order Zipcode Product Card Id Product Category Id  Product Description  \
1                NaN            1360                  73                  NaN   
21               NaN            1360                  73                  NaN   
45               NaN            1360                  73                  NaN   
46               NaN            1360                  73                  NaN   
382              NaN            1360                  73                  NaN   
...              ...             ...                 ...                  ...   
180142           NaN            1004                  45                  NaN   
180164           NaN            1004                  45                  NaN   
180248           NaN            1004                  45                  NaN   
180331           NaN            1004                  45                  NaN   
180405           NaN            1004                  45                  NaN   

                                            Product Image  \
1            http://images.acmesports.sports/Smart+watch    
21           http://images.acmesports.sports/Smart+watch    
45           http://images.acmesports.sports/Smart+watch    
46           http://images.acmesports.sports/Smart+watch    
382          http://images.acmesports.sport

In [15]:
print(pending_indian_dataframe["Order City"].unique())

['Bikaner' 'Jabalpur' 'Bangalore' 'Pune' 'Jaipur' 'Ranchi' 'Durg' 'Patna'
 'Faridabad' 'Fatehpur' 'Kota' 'Guwahati' 'Rohtak' 'Mumbai' 'Kulti'
 'Imphal' 'Gorakhpur' 'Lucknow' 'Hubli' 'Nagpur' 'Ajmer' 'Amravati'
 'Raipur' 'Thane' 'Dewas' 'Tirunelveli' 'Darbhanga' 'Aligarh' 'Saharanpur'
 'Vadodara' 'Delhi' 'Bhopal' 'Jodhpur' 'Kanpur' 'Moradabad' 'Belgaum'
 'Hyderabad' 'Nellore' 'Korba' 'Jamnagar' 'Puducherry' 'Srinagar' 'Satna'
 'Muzaffarpur' 'Pimpri' 'Bokaro' 'Ludhiana' 'Jhansi' 'Amritsar' 'Malegaon'
 'Nasik' 'Coimbatore' 'Agra' 'Durgapur' 'Chennai' 'Gulbarga' 'Brahmapur'
 'Thiruvananthapuram' 'Bhavnagar' 'Ujjain' 'Surat' 'Ganganagar'
 'Visakhapatnam' 'Solapur' 'Salem' 'Meerut' 'Patiala' 'Rajkot' 'Warangal'
 'Kollam' 'Hisar' 'Mangalore' 'Aurangabad' 'Bhubaneswar' 'Bilaspur'
 'Kamarhati' 'Ulhasnagar' 'Shimoga' 'Parbhani' 'Mysore' 'Bhiwandi'
 'Bellary' 'Anantapur' 'Bihar Sharif' 'Ghaziabad' 'Madurai' 'Ichalkaranji'
 'Tiruppur' 'Kolhapur' 'Chandigarh' 'Jammu']


In [16]:
a = pending_indian_dataframe[pending_indian_dataframe['Order City'].value_counts()]

In [68]:
type(a)
a

Kanpur        25
Pune          24
Bangalore     24
Lucknow       21
Mumbai        18
              ..
Patiala        1
Aurangabad     1
Ghaziabad      1
Hubli          1
Jamnagar       1
Name: Order City, Length: 91, dtype: int64

In [67]:
print(a.head(50))
print(a.tail(50))
#all the cities in India with the number of pending orders

Kanpur           25
Pune             24
Bangalore        24
Lucknow          21
Mumbai           18
Delhi            18
Hyderabad        17
Ajmer            17
Visakhapatnam    15
Korba            14
Nagpur           13
Chennai          12
Gorakhpur        12
Jabalpur         11
Durgapur         11
Faridabad        10
Darbhanga         9
Jaipur            9
Raipur            8
Guwahati          8
Kulti             8
Belgaum           8
Surat             7
Salem             7
Patna             7
Kota              7
Agra              7
Jodhpur           7
Bellary           6
Bhubaneswar       6
Nasik             6
Amritsar          6
Dewas             6
Ujjain            5
Imphal            5
Coimbatore        5
Moradabad         5
Tirunelveli       5
Ranchi            5
Bokaro            5
Ulhasnagar        5
Rohtak            5
Satna             5
Jhansi            4
Durg              4
Mysore            4
Malegaon          4
Srinagar          4
Aligarh           4
Saharanpur        4


# Ans. The above given cities of India are having pending orders(numbers given)

# 2. In this part we will be trying to find the Country with the most Suspected Fraud.

In [21]:
fraud_dataframe = df[df['Order Status'] == 'SUSPECTED_FRAUD']

In [23]:
fraud_dataframe
#this dataframe contains all the orders having Suspected_Fraud as its status.

Type  Days for shipping (real)  Days for shipment (scheduled)  \
10      TRANSFER                         6                              2   
24      TRANSFER                         2                              2   
183     TRANSFER                         5                              4   
184     TRANSFER                         5                              4   
185     TRANSFER                         6                              4   
...          ...                       ...                            ...   
180274  TRANSFER                         5                              2   
180309  TRANSFER                         2                              1   
180352  TRANSFER                         4                              4   
180406  TRANSFER                         6                              4   
180490  TRANSFER                         4                              4   

        Benefit per order  Sales per customer    Delivery Status  \
10             130.580002          272.029999  Shipping canceled   
24              90.279999          288.420013  Shipping canceled   
183             28.850000          128.220001  Shipping canceled   
184            133.910004          278.970001  Shipping canceled   
185             79.160004          272.970001  Shipping canceled   
...                   ...                 ...                ...   
180274         163.190002          339.980011  Shipping canceled   
180309         114.830002          347.980011  Shipping canceled   
180352          90.000000          399.980011  Shipping canceled   
180406        -333.179993          399.980011  Shipping canceled   
180490         -59.840000          339.980011  Shipping canceled   

        Late_delivery_risk  Category Id     Category Name    Customer City  \
10                       0           73    Sporting Goods           Caguas   
24                       0           73    Sporting Goods         Billings   
183                      0           13       Electronics         Freeport   
184                      0            9  Cardio Equipment  Fort Washington   
185                      0            9  Cardio Equipment      Bakersfield   
...                    ...          ...               ...              ...   
180274                   0           45           Fishing           Caguas   
180309                   0           45           Fishing          Modesto   
180352                   0           45           Fishing      Albuquerque   
180406                   0           45           Fishing           Caguas   
180490                   0           45           Fishing           Madera   

        ... Order Zipcode Product Card Id Product Category Id  \
10      ...           NaN            1360                  73   
24      ...           NaN            1360                  73   
183     ...       77041.0             278                  13   
184     ...       85234.0             191                   9   
185     ...       95123.0             191                   9   
...     ...           ...             ...                 ...   
180274  ...           NaN            1004                  45   
180309  ...           NaN            1004                  45   
180352  ...           NaN            1004                  45   
180406  ...           NaN            1004                  45   
180490  ...           NaN            1004                  45   

        Product Description  \
10                      NaN   
24                      NaN   
183                     NaN   
184                     NaN   
185                     NaN   
...                     ...   
180274                  NaN   
180309                  NaN   
180352                  NaN   
180406                  NaN   
180490                  NaN   

                                            Product Image  \
10           http://images.acmesports.sports/Smart+watch    
24           http://images.acmesports.sports/Smart+watch 

In [25]:
fraud_dataframe['Order Country'].value_counts()

Estados Unidos    573
Francia           362
México            317
Alemania          227
Australia         197
                 ... 
Bangladés           1
Somalia             1
Benín               1
Uzbekistán          1
Georgia             1
Name: Order Country, Length: 102, dtype: int64

# Ans. From the above cell's output we can see that "Estados Unidos" has the most suspected fraud in order status than other countries.

# 3. Which product(s) has/have the maximum discount?

In [26]:
print(updated_df["Product Name"].unique())

['Smart watch ' 'Perfect Fitness Perfect Rip Deck'
 "Under Armour Girls' Toddler Spine Surge Runni"
 "Nike Men's Dri-FIT Victory Golf Polo"
 "Under Armour Men's Compression EV SL Slide"
 "Under Armour Women's Micro G Skulpt Running S"
 "Nike Men's Free 5.0+ Running Shoe"
 "Glove It Women's Mod Oval 3-Zip Carry All Gol"
 'Bridgestone e6 Straight Distance NFL San Dieg'
 "Columbia Men's PFG Anchor Tough T-Shirt" 'Titleist Pro V1x Golf Balls'
 'Bridgestone e6 Straight Distance NFL Tennesse'
 'Polar FT4 Heart Rate Monitor' 'ENO Atlas Hammock Straps'
 "adidas Men's F10 Messi TRX FG Soccer Cleat"
 "Brooks Women's Ghost 6 Running Shoe"
 "Nike Men's CJ Elite 2 TD Football Cleat"
 "Diamondback Women's Serene Classic Comfort Bi"
 'Industrial consumer electronics' 'Web Camera' 'Dell Laptop'
 'SOLE E25 Elliptical' 'Elevation Training Mask 2.0'
 "adidas Men's Germany Black Crest Away Tee"
 'Team Golf Pittsburgh Steelers Putter Grip'
 'Glove It Urban Brick Golf Towel' 'Team Golf Texas Longhorns Putte

In [27]:
print(updated_df["Order Item Discount"].unique())

[ 13.10999966  16.38999939  18.03000069 ... 140.          33.65999985
  61.18999863]


In [31]:
discount_column = updated_df["Order Item Discount"]
max_value = discount_column.max()
print(max_value)

500.0


In [32]:
max_discount_dataframe = updated_df[updated_df['Order Item Discount'] == 500.0]

In [33]:
max_discount_dataframe

Type  Days for shipping (real)  Days for shipment (scheduled)  \
12606  DEBIT                         4                              4   

       Benefit per order  Sales per customer   Delivery Status  \
12606               45.0          1499.98999  Shipping on time   

       Late_delivery_risk  Category Id      Category Name Customer City  ...  \
12606                   0           10  Strength Training        Caguas  ...   

      Order Status Order Zipcode Product Card Id  Product Category Id  \
12606     COMPLETE           NaN             208                   10   

                                           Product Image         Product Name  \
12606  http://images.acmesports.sports/SOLE+E35+Ellip...  SOLE E35 Elliptical   

      Product Price Product Status shipping date (DateOrders)   Shipping Mode  
12606    1999.98999              0            10/3/2017 21:22  Standard Class  

[1 rows x 52 columns]

In [35]:
print(updated_df.loc[updated_df['Order Item Discount'] == 500.0])

        Type  Days for shipping (real)  Days for shipment (scheduled)  \
12606  DEBIT                         4                              4   

       Benefit per order  Sales per customer   Delivery Status  \
12606               45.0          1499.98999  Shipping on time   

       Late_delivery_risk  Category Id      Category Name Customer City  ...  \
12606                   0           10  Strength Training        Caguas  ...   

      Order Status Order Zipcode Product Card Id  Product Category Id  \
12606     COMPLETE           NaN             208                   10   

                                           Product Image         Product Name  \
12606  http://images.acmesports.sports/SOLE+E35+Ellip...  SOLE E35 Elliptical   

      Product Price Product Status shipping date (DateOrders)   Shipping Mode  
12606    1999.98999              0            10/3/2017 21:22  Standard Class  

[1 rows x 52 columns]


In [55]:
print(max_discount_dataframe['Product Name'].head())
#this gives us the name of product(s) which has/have the maximum discount

12606    SOLE E35 Elliptical
Name: Product Name, dtype: object


# Ans. The product with name SOLE E35 Elliptical has maximum discount.

# 5. In this part we will find the How many successfully processed orders have days for shipping real (Real) > Days for shipment (Scheduled)?

In [56]:
Completed_order_dataframe = updated_df[updated_df['Order Status'] == 'COMPLETE']

In [57]:
Completed_order_dataframe

Type  Days for shipping (real)  Days for shipment (scheduled)  \
0       DEBIT                         3                              4   
3       DEBIT                         3                              4   
6       DEBIT                         2                              1   
13      DEBIT                         2                              1   
15      DEBIT                         2                              1   
...       ...                       ...                            ...   
180500  DEBIT                         3                              4   
180505  DEBIT                         2                              4   
180508  DEBIT                         4                              2   
180512  DEBIT                         6                              4   
180515  DEBIT                         3                              2   

        Benefit per order  Sales per customer   Delivery Status  \
0               91.250000          314.640015  Advance shipping   
3               22.860001          304.809998  Advance shipping   
6               95.180000          288.420013     Late delivery   
13              24.580000          245.809998     Late delivery   
15            -259.579987          324.470001     Late delivery   
...                   ...                 ...               ...   
180500         138.229996          383.980011  Advance shipping   
180505         112.669998          359.980011  Advance shipping   
180508          85.000000          339.980011     Late delivery   
180512          86.400002          319.980011     Late delivery   
180515        -613.770019          395.980011     Late delivery   

        Late_delivery_risk  Category Id   Category Name Customer City  ...  \
0                        0           73  Sporting Goods        Caguas  ...   
3                        0           73  Sporting Goods   Los Angeles  ...   
6                        1           73  Sporting Goods        Caguas  ...   
13                       1           73  Sporting Goods        Caguas  ...   
15                       1           73  Sporting Goods        Caguas  ...   
...                    ...          ...             ...           ...  ...   
180500                   0           45         Fishing    Saint Paul  ...   
180505                   0           45         Fishing      Highland  ...   
180508                   1           45         Fishing       Waipahu  ...   
180512                   1           45         Fishing        Caguas  ...   
180515                   1           45         Fishing   Bakersfield  ...   

       Order Status Order Zipcode Product Card Id  Product Category Id  \
0          COMPLETE           NaN            1360                   73   
3          COMPLETE           NaN            1360                   73   
6          COMPLETE           NaN            1360                   73   
13         COMPLETE           NaN            1360                   73   
15         COMPLETE           NaN            1360                   73   
...             ...           ...             ...                  ...   
180500     COMPLETE           NaN            1004                   45   
180505     COMPLETE           NaN            1004                   45   
180508     COMPLETE           NaN            1004                   45   
180512     COMPLETE           NaN            1004                   45   
180515     COMPLETE           NaN            1004                   45   

                                            Product Image  \
0            http://images.acmesports.sports/Smart+watch    
3            http://images.acmesports.sports/Smart+watch    
6            http://images.acmesports.sports/Smart+watch    
13           http://images.acmesports.sports/Smart+watch    
15           http://images.acmesports.sports/Smart+watch    
...                                                   ...   
180500  http://images.acmesports.sports/Field+%26+Stre...   
1

In [61]:
Schedules = Completed_order_dataframe[Completed_order_dataframe["Days for shipping (real)"] > Completed_order_dataframe["Days for shipment (scheduled)"]]

In [65]:
Schedules
#this gives us a datframe with the days for shipping real (Real) > Days for shipment (Scheduled) condition.


Type  Days for shipping (real)  Days for shipment (scheduled)  \
6       DEBIT                         2                              1   
13      DEBIT                         2                              1   
15      DEBIT                         2                              1   
18      DEBIT                         2                              1   
25      DEBIT                         6                              2   
...       ...                       ...                            ...   
180479  DEBIT                         6                              4   
180499  DEBIT                         4                              2   
180508  DEBIT                         4                              2   
180512  DEBIT                         6                              4   
180515  DEBIT                         3                              2   

        Benefit per order  Sales per customer Delivery Status  \
6               95.180000          288.420013   Late delivery   
13              24.580000          245.809998   Late delivery   
15            -259.579987          324.470001   Late delivery   
18             102.260002          314.640015   Late delivery   
25             131.169998          285.140015   Late delivery   
...                   ...                 ...             ...   
180479         133.839996          395.980011   Late delivery   
180499        -258.779999          387.980011   Late delivery   
180508          85.000000          339.980011   Late delivery   
180512          86.400002          319.980011   Late delivery   
180515        -613.770019          395.980011   Late delivery   

        Late_delivery_risk  Category Id   Category Name Customer City  ...  \
6                        1           73  Sporting Goods        Caguas  ...   
13                       1           73  Sporting Goods        Caguas  ...   
15                       1           73  Sporting Goods        Caguas  ...   
18                       1           73  Sporting Goods        Caguas  ...   
25                       1           73  Sporting Goods        Caguas  ...   
...                    ...          ...             ...           ...  ...   
180479                   1           45         Fishing     Las Vegas  ...   
180499                   1           45         Fishing        Caguas  ...   
180508                   1           45         Fishing       Waipahu  ...   
180512                   1           45         Fishing        Caguas  ...   
180515                   1           45         Fishing   Bakersfield  ...   

       Order Status Order Zipcode Product Card Id  Product Category Id  \
6          COMPLETE           NaN            1360                   73   
13         COMPLETE           NaN            1360                   73   
15         COMPLETE           NaN            1360                   73   
18         COMPLETE           NaN            1360                   73   
25         COMPLETE           NaN            1360                   73   
...             ...           ...             ...                  ...   
180479     COMPLETE           NaN            1004                   45   
180499     COMPLETE           NaN            1004                   45   
180508     COMPLETE           NaN            1004                   45   
180512     COMPLETE           NaN            1004                   45   
180515     COMPLETE           NaN            1004                   45   

                                            Product Image  \
6            http://images.acmesports.sports/Smart+watch    
13           http://images.acmesports.sports/Smart+watch    
15           http://images.acmesports.sports/Smart+watch    
18           http://images.acmesports.sports/Smart+watch    
25           http://images.acmesports.sports/Smart+watch    
...                                                   ...   
180479  http://images.acmesports.sports/Field+%26+Stre...   
180499  http://images.acm

In [64]:
Schedules.shape

(34199, 52)

# ans. From the above output we see that there are 34199 orders that have days for shipping real (Real) > Days for shipment (Scheduled).

# 4. Which year had the minimum sales for "Nike" products?

In [95]:
nike_dataframe = updated_df.loc[updated_df['Product Name'].str.contains("nike", case=False)]

In [96]:
nike_dataframe

Type  Days for shipping (real)  Days for shipment (scheduled)  \
50       PAYMENT                         6                              2   
52       PAYMENT                         3                              2   
53       PAYMENT                         5                              2   
54       PAYMENT                         6                              2   
59       PAYMENT                         2                              2   
...          ...                       ...                            ...   
179604  TRANSFER                         5                              4   
179605  TRANSFER                         2                              4   
179606  TRANSFER                         3                              4   
179607  TRANSFER                         5                              4   
179608  TRANSFER                         4                              4   

        Benefit per order  Sales per customer   Delivery Status  \
50              33.599998           96.000000     Late delivery   
52               9.100000           91.000000     Late delivery   
53             -21.750000           87.000000     Late delivery   
54               6.150000           82.000000     Late delivery   
59            -184.779999          263.970001  Shipping on time   
...                   ...                 ...               ...   
179604         -33.750000          449.959992     Late delivery   
179605         141.360001          434.959992  Advance shipping   
179606         188.979996          419.959992  Advance shipping   
179607         111.989998          399.959992     Late delivery   
179608         167.979996          399.959992  Shipping on time   

        Late_delivery_risk  Category Id     Category Name Customer City  ...  \
50                       1           24   Women's Apparel        Caguas  ...   
52                       1           24   Women's Apparel        Caguas  ...   
53                       1           24   Women's Apparel        Caguas  ...   
54                       1           24   Women's Apparel        Caguas  ...   
59                       0            9  Cardio Equipment        Caguas  ...   
...                    ...          ...               ...           ...  ...   
179604                   1            9  Cardio Equipment        Caguas  ...   
179605                   0            9  Cardio Equipment        Caguas  ...   
179606                   0            9  Cardio Equipment        Caguas  ...   
179607                   1            9  Cardio Equipment        Caguas  ...   
179608                   0            9  Cardio Equipment        Caguas  ...   

           Order Status Order Zipcode Product Card Id  Product Category Id  \
50      PENDING_PAYMENT           NaN             502                   24   
52      PENDING_PAYMENT           NaN             502                   24   
53      PENDING_PAYMENT           NaN             502                   24   
54      PENDING_PAYMENT           NaN             502                   24   
59      PENDING_PAYMENT           NaN             191                    9   
...                 ...           ...             ...                  ...   
179604          PENDING           NaN             191                    9   
179605          PENDING           NaN             191                    9   
179606          PENDING           NaN             191                    9   
179607          PENDING           NaN             191                    9   
179608          PENDING           NaN             191                    9   

                                            Product Image  \
50      http://images.acmesports.sports/Nike+Men%27s+D...   
52      http://images.acmesports.sports/Nike+Men%27s+D...   
53      http://images.acmesports.sports/Nike+Men%27s+D...   
54      http://images.acmesports.sports/Nike+Men%27s+D...   
59      http://images.acmesports.sports/Nike+Men%27s+F...   
...               

In [102]:
nike2015_dataframe = updated_df.loc[updated_df['shipping date (DateOrders)'].str.contains("2015", case=False)]
sales_per_cust_column = nike2015_dataframe["Order Item Discount"]
sum_value2015 = sales_per_cust_column.sum()

In [103]:
sum_value2015
#the sales in year 2015

1238527.2911785059

In [104]:
nike2016_dataframe = updated_df.loc[updated_df['shipping date (DateOrders)'].str.contains("2016", case=False)]
sales_per_cust_column = nike2016_dataframe["Order Item Discount"]
sum_value2016 = sales_per_cust_column.sum()

In [105]:
sum_value2016
#the sales in year 2016

1247833.721141224

In [106]:
nike2017_dataframe = updated_df.loc[updated_df['shipping date (DateOrders)'].str.contains("2017", case=False)]
sales_per_cust_column = nike2017_dataframe["Order Item Discount"]
sum_value2017 = sales_per_cust_column.sum()

In [107]:
sum_value2017
#the sales in year 2017

1202255.410429992

In [108]:
nike2018_dataframe = updated_df.loc[updated_df['shipping date (DateOrders)'].str.contains("2018", case=False)]
sales_per_cust_column = nike2018_dataframe["Order Item Discount"]
sum_value2018 = sales_per_cust_column.sum()

In [109]:
sum_value2018
#the sales in year 2018

41761.979842847

In [110]:
#as we can see from the above values nike had minimum sales in 2018

# Nike products had minimum sales in the year 2018